### PR TITLE
Set the max number of driver instances to CPU cores

### PR DESCRIPTION
**Pull request by vmarkovtsev**
+ Add the environment variable

### CODE REVIEW

File: pool.go [hunk 1]
```diff
@@ -235,7 +239,20 @@ type ScalingPolicy interface {
 // DefaultScalingPolicy returns a new instance of the default scaling policy.
 // Instances returned by this function should not be reused.
 func DefaultScalingPolicy() ScalingPolicy {
-	return MovingAverage(10, MinMax(1, 10, AIMD(1, 0.5)))
+	maxInstances := 0
+	// Try to read maxInstances from the environment variable
+	maxInstancesEnv := os.Getenv("BBLFSH_SCALING_MAX_INSTANCES")
```
Comment (abeaumont) [inline pool.go]: I think it'd be better to move this context dependant code outside and pass them as arguments explicitely

Comment (vmarkovtsev) [inline pool.go]: I see that `DefaultScalingPolicy()` is used in many places, including the tests. What shall I do with all of those?

Comment (smola) [inline pool.go]: @abeaumont that can be done by not using `DefaultScalingPolicy()` and instantiated it instead.

Nonetheless, this code should be in `func init()`, initializing a global variable (e.g. `DefaulMaxInstancesPerDriver`), and then using that variable here.

But you're right that at some point we should move this kind of configuration to `server.go` and propagate it up to the `cli` package. I think would leave that to a further PR though.

**Comment by vmarkovtsev** [comment]
@smola Did as you said, changed the corresponding test to pass.

**Pull request merged**